src/modm/driver/can/mcp2515_impl.hpp

```diff
@@ -312,128 +421,143 @@ template <typename SPI, typename CS, typename INT>
 void
 modm::Mcp2515<SPI, CS, INT>::writeRegister(uint8_t address, uint8_t data)
 {
+	while(!this->acquireMaster()){};
```

Comment:
these functions are called from initialize(), which should probably be blocking (I kept it blocking)

Reply:
Imagine the following situation: You use the `SpiMaster` with another driver and then initialize the Mcp2515 while another transaction is running from the other device. Then the `initialize()` will deadlock forever. Even if that situation is not that common, it will cause nasty bugs.

If I am not mistaken every other initialization function that writes SPI / I²C in modm drivers is resumable. If something should be forced to execute blocking there is always `RF_CALL_BLOCKING()`.

Reply:
Yeah I am aware of that limitation .. I am still playing around with other stuff so I will try to get the init stuff properly set up later :) just have to do other things for a while :D